python/tests/test_account.py

```diff
@@ -1337,18 +1338,31 @@ def ac_outgoing_message(self, message):
         m = message_queue.get()
         assert m == msg_in
 
-    def test_import_export_online_all(self, acfactory, tmpdir, lp):
+    def test_import_export_online_all(self, acfactory, tmpdir, data, lp):
         ac1 = acfactory.get_one_online_account()
 
         lp.sec("create some chat content")
-        contact1 = ac1.create_contact("some1@example.org", name="some1")
-        contact1.create_chat().send_text("msg1")
+        chat1 = ac1.create_contact("some1@example.org", name="some1").create_chat()
+        chat1.send_text("msg1")
         assert len(ac1.get_contacts(query="some1")) == 1
+
+        original_image_path = data.get_path("d.png")
+        chat1.send_image(original_image_path)
+
         backupdir = tmpdir.mkdir("backup")
 
         lp.sec("export all to {}".format(backupdir))
-        path = ac1.export_all(backupdir.strpath)
-        assert os.path.exists(path)
+        with ac1.temp_plugin(ImexTracker()) as imex_tracker:
+            path = ac1.export_all(backupdir.strpath)
+            assert os.path.exists(path)
+
+            # check progress events for export
+            assert imex_tracker.wait_progress(1, progress_upper_limit=249)
```

Comment:
This assert only checks that returned value is not 0 or `None`. Maybe check that it actually returns `1` (and not `2` or anything like that)?

Reply:
In this line we test that some number between 1 and 249 is emitted at some point (just that the logic for this is in `wait_progress()`). That's how fine-grained this test can be because only 4 progresses are emitted.

That's because only after successfully saving a file a progress is emitted, so to get more progress events we would need to send more files.